service/access.go

```diff
@@ -51,7 +54,8 @@ type accessService struct {
 	token                 ntokend.TokenProvider
 	athenzURL             string
 	athenzPrincipleHeader string
-	tokenCache            gache.Gache
+	tokenCache            gache.Gache[accessCacheData]
```

Comment:
Previous map store pointer as map value. Is there any result to use the struct as the map value?